### PR TITLE
[WIP] Removes .toJS()

### PR DESCRIPTION
**Pull request by lukaszsobek**
Fixes Issue: #4837

Removes all 'toJS()' calls for performance resons.

### CODE REVIEW

File: src/actions/breakpoints/addBreakpoint.js [hunk 1]
```diff
@@ -51,7 +53,11 @@ export default async function addBreakpoint(
   );
 
   const symbols = getSymbols(getState(), sourceRecord);
-  const astLocation = await getASTLocation(sourceRecord, symbols, newLocation);
+  const astLocation = await getASTLocation(
+    sourceRecord.toJS(),
```
Comment (lukaszsobek) [inline src/actions/breakpoints/addBreakpoint.js]: Temporary override until I can focus on ast locations.

File: src/actions/ast.js [hunk 2]
```diff
@@ -25,19 +25,17 @@ import type { ThunkArgs } from "./types";
 export function setSourceMetaData(sourceId: SourceId) {
   return async ({ dispatch, getState }: ThunkArgs) => {
     const sourceRecord = getSource(getState(), sourceId);
-    if (!sourceRecord) {
+    if (
+      !sourceRecord ||
+      !sourceRecord.get("text") ||
+      sourceRecord.get("isWasm")
```
Comment (jasonLaster) [inline src/actions/ast.js]: this can now be

`sourceRecord.text` and `sourceRecord.isWasm`

File: src/actions/ast.js [hunk 3]
```diff
@@ -25,19 +25,17 @@ import type { ThunkArgs } from "./types";
...
       return;
     }
-
-    const source = sourceRecord.toJS();
-    if (!source.text || source.isWasm) {
-      return;
-    }
-
-    const framework = await getFramework(source.id);
+    const framework = await getFramework(sourceRecord.get("id"));
```
Comment (jasonLaster) [inline src/actions/ast.js]: `sourceRecord.id`

File: src/actions/ast.js [hunk 4]
```diff
@@ -48,17 +46,16 @@ export function setSourceMetaData(sourceId: SourceId) {
 export function setSymbols(sourceId: SourceId) {
   return async ({ dispatch, getState }: ThunkArgs) => {
     const sourceRecord = getSource(getState(), sourceId);
-    if (!sourceRecord) {
+    if (
+      !sourceRecord ||
+      !sourceRecord.get("text") ||
+      sourceRecord.get("isWasm") ||
```
Comment (jasonLaster) [inline src/actions/ast.js]: same here

File: src/components/Editor/EditorMenu.js [hunk 5]
```diff
@@ -226,20 +226,20 @@ const {
...
 export default connect(
   state => {
-    const selectedSource = getSelectedSource(state);
+    const selectedSourceRecord = getSelectedSource(state);
     return {
       selectedLocation: getSelectedLocation(state),
-      selectedSource,
-      hasPrettyPrint: !!getPrettySource(state, selectedSource.get("id")),
+      selectedSourceRecord,
+      hasPrettyPrint: !!getPrettySource(state, selectedSourceRecord.get("id")),
```
Comment (jasonLaster) [inline src/components/Editor/EditorMenu.js]: selectedSourceRecord.id

**Comment by jasonLaster** [comment]
closing as we'll open some smaller PRs

**Pull request closed without merging**